### PR TITLE
Fix shouldHideReaderRevenue javascript config

### PR DESCRIPTION
**Pull request by jranks123**
## What does this change?

This PR is a follow up to https://github.com/guardian/frontend/pull/19228 and https://github.com/guardian/frontend/pull/19207.  The former was fixing a bug where the `shouldHideReaderRevenue` config attribute was not defaulting to true for paid content. The latter was making shouldHideReaderRevenue a top level page property. 

The problem with the latter's implementation was that it read the shouldHideReaderRevenue straight from the API response, which ignores the logic that goes on in the `shouldHideReaderRevenue` function in  the `Fields` object in `meta.scala`, which reads the flag then applies some additional logic for articles published before a cutoff date, and for paid content. 

Seeing as the `fields` object already contains `shouldHideReaderRevenue`, we do not need to define it again here, so we can remove it. 

When composing the config for content pages, the `shouldHideReaderRevenue` flag was being set by the `content` javascript content, then being overridden by the `readerRevenueMetaData`. To make it simpler, these have both been removed and the `shouldHideReaderRevenue` page property is being set in the `Fields` case class, where it is originally calculated with the correct logic. 

## What is the value of this and can you measure success?
The `shouldHideReaderRevenue` flag is set correctly, and we will no longer show our ads on paid content. 

## Does this affect other platforms - Amp, Apps, etc?
No

## Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

Yes

### CODE REVIEW

File: common/app/model/content.scala [hunk 1]
```diff
@@ -389,7 +388,7 @@ object Content {
       showInRelated = apifields.flatMap(_.showInRelatedContent).getOrElse(false),
       cardStyle = CardStyle.make(cardStyle),
       shouldHideAdverts = apifields.flatMap(_.shouldHideAdverts).getOrElse(false),
-      shouldHideReaderRevenue = apifields.flatMap(_.shouldHideReaderRevenue).getOrElse(false),
+      shouldHideReaderRevenue = fields.shouldHideReaderRevenue.getOrElse(false),
```
Comment (philmcmahon) [inline common/app/model/content.scala]: is this still necessary? if it's in Fields maybe we don't need it in Content?

**Comment by philmcmahon** [comment]
Other than comment above, 👍

File: common/app/model/content.scala (hunk 1 above)
Comment (jranks123) [reply common/app/model/content.scala]: 👍

**Review by philmcmahon — APPROVED** [review]

**Review by joelochlann — COMMENTED** [review]
Could we add a test to prevent this happening again? I created a test for the actual `shouldHideReaderRevenue` function (see https://github.com/guardian/frontend/blob/master/common/test/model/MetaDataTest.scala). But we don't have one that checks if it's actually in the JavaScript config for various content types.

File: common/test/model/ContentTest.scala [hunk 2]
```diff
@@ -132,6 +138,56 @@ class ContentTest extends FlatSpec with Matchers with GuiceOneAppPerSuite with i
...
+    JavaScriptPage.getMap(SimpleContentPage(content), edition, false).get("shouldHideReaderRevenue") should be(Some(JsBoolean(true)))
+  }
+
+  it should "detect that an old, non sensitive article should not hide reader revenue" in {
+
+    val content =  Content(article.copy(webPublicationDate = Some(dateBeforeCutoff.toCapiDateTime), fields = Some(ContentFields(sensitive = Some(false)))))
+    JavaScriptPage.getMap(SimpleContentPage(content), edition, false).get("shouldHideReaderRevenue") should be(Some(JsBoolean(false)))
+  }
+
+  it should "detect that an new, non sensitive article should not hide reader revenue if the shouldHideReaderRevenue flag is set to false" in {
```
Comment (joelochlann) [inline common/test/model/ContentTest.scala]: typo: an new

File: common/test/model/ContentTest.scala [hunk 3]
```diff
@@ -132,6 +138,56 @@ class ContentTest extends FlatSpec with Matchers with GuiceOneAppPerSuite with i
...
+
+    val content = Content(article.copy(webPublicationDate = Some(dateBeforeCutoff.toCapiDateTime), fields = Some(ContentFields(sensitive = Some(false), shouldHideReaderRevenue = Some(false)))))
+    JavaScriptPage.getMap(SimpleContentPage(content), edition, false).get("shouldHideReaderRevenue") should be(Some(JsBoolean(false)))
+  }
+
+
+  it should "detect that an new, non sensitive article should hide reader revenue if the shouldHideReaderRevenue flag is set to true" in {
```
Comment (joelochlann) [inline common/test/model/ContentTest.scala]: an new

File: common/test/model/ContentTest.scala [hunk 4]
```diff
@@ -141,6 +147,59 @@ class ContentTest extends FlatSpec with Matchers with GuiceOneAppPerSuite with i
...
+  val dateBeforeCutoff = Some(jodaToJavaInstant(new DateTime("2017-07-02T12:00:00.000Z")).atOffset(ZoneOffset.UTC).toCapiDateTime)
+  val dateAfterCutoff = Some(jodaToJavaInstant(new DateTime("2017-07-04T12:00:00.000Z")).atOffset(ZoneOffset.UTC).toCapiDateTime)
+  val edition: Edition = mock[Edition]
+  val sensitiveContentFields = Some(ContentFields(sensitive = Some(true)))
+  val nonSensitiveContentFields = Some(ContentFields(sensitive = Some(true)))
+
+
+  when(edition.id) thenReturn "GB"
+
+  "ContentJavascriptConfig" should "set shouldHideReaderRevenue to true for a sensitive article, published before the cutoff date, no RR flag" in {
```
Comment (joelochlann) [inline common/test/model/ContentTest.scala]: spacing's a bit weird, looks like `ContentJavascriptConfig` is a variable name - maybe "Content JavaScript config"

**Review by joelochlann — APPROVED** [review]
Thanks Jonny

**Pull request merged**